crates/services/block_aggregator_api/src/api/protobuf_adapter/tests.rs

```diff
@@ -35,76 +32,52 @@ use futures::{
     StreamExt,
     TryStreamExt,
 };
+use std::{
+    net::SocketAddr,
+    sync::Arc,
+};
+use tokio::sync::broadcast;
 
-fn free_local_addr() -> String {
+fn free_local_addr() -> SocketAddr {
     let listener = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
-    let addr = listener.local_addr().unwrap();
-    format!("127.0.0.1:{}", addr.port())
+    listener.local_addr().unwrap()
 }
 
 #[tokio::test]
 async fn await_query__get_current_height__client_receives_expected_value() {
-    // given
-    let path = free_local_addr();
-    let mut api = ProtobufAPI::new(path.to_string(), 100).unwrap();
-    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+    let socket = free_local_addr();
+    let mut api = MockBlocksAggregatorApi::default();
+
+    // Given
+    api.expect_get_current_height()
+        .times(1)
```

Comment:
This is why I don't like `mockall`. We don't care how many times it is called--it's over-constrained.

i.e. it's like we have an extra `// then` that isn't in the `// then` section.